crates/types/src/bindings.rs

```diff
@@ -1,118 +1,146 @@
 #![allow(clippy::too_many_arguments)]
 #![allow(missing_docs)]
-use alloy_primitives::Address;
+use alloy_primitives::{Address, Bytes, FixedBytes, U256};
 use alloy_sol_types::sol;
 
-use self::Orders::{OrdersErrors, OrdersEvents};
-
 sol!(
     #[sol(rpc)]
     #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
     Zenith,
     "abi/Zenith.json"
 );
 
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    Passage,
+    "abi/Passage.json"
+);
+
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    RollupOrders,
+    "abi/RollupOrders.json"
+);
+
 // Zenith types
 impl Copy for Zenith::BlockHeader {}
 impl Copy for Zenith::BlockSubmitted {}
 impl Copy for Zenith::SequencerSet {}
-impl Copy for Zenith::BadSequence {}
 impl Copy for Zenith::BadSignature {}
-impl Copy for Zenith::BlockExpired {}
 impl Copy for Zenith::OneRollupBlockPerHostBlock {}
 impl Copy for Zenith::OnlySequencerAdmin {}
+impl Copy for Zenith::IncorrectHostBlock {}
 
-// Passage types
-impl Copy for Zenith::Enter {}
-impl Copy for Zenith::Withdrawal {}
-impl Copy for Zenith::OnlyWithdrawalAdmin {}
-
-impl Copy for Zenith::ZenithErrors {}
-
-impl Clone for Zenith::ZenithErrors {
-    fn clone(&self) -> Self {
-        *self
+impl Zenith::BlockSubmitted {
+    pub const fn sequencer(&self) -> Address {
+        self.sequencer
     }
-}
-
-impl Clone for Zenith::ZenithEvents {
-    fn clone(&self) -> Self {
-        match self {
-            Self::BlockSubmitted(inner) => Self::BlockSubmitted(*inner),
-            Self::Enter(inner) => Self::Enter(*inner),
-            Self::SequencerSet(inner) => Self::SequencerSet(*inner),
-            Self::Transact(inner) => Self::Transact(inner.clone()),
-            Self::Withdrawal(inner) => Self::Withdrawal(*inner),
-        }
+    pub const fn rollup_chain_id(&self) -> u64 {
+        self.rollupChainId.as_limbs()[0]
+    }
+    pub const fn gas_limit(&self) -> u64 {
+        self.gasLimit.as_limbs()[0]
+    }
+    pub const fn reward_address(&self) -> Address {
+        self.rewardAddress
+    }
+    pub const fn block_data_hash(&self) -> FixedBytes<32> {
+        self.blockDataHash
     }
 }
 
-impl From<&Zenith::BlockSubmitted> for Zenith::BlockHeader {
-    fn from(event: &Zenith::BlockSubmitted) -> Zenith::BlockHeader {
-        Zenith::BlockHeader {
-            rollupChainId: event.rollupChainId,
-            sequence: event.sequence,
-            confirmBy: event.confirmBy,
-            gasLimit: event.gasLimit,
-            rewardAddress: event.rewardAddress,
-            blockDataHash: event.blockDataHash,
-        }
+// returns a BlockHeader from a BlockSubmitted event with the given host block number
+#[allow(dead_code)]
+pub(crate) const fn header_from_block_submitted(
```

Comment:
this can be `pub` instead of `pub(crate)`

should also be inside the impl block above. so it's accessed as `Zenith::BlockHeeader::from_block_submitted(...)` instea of  a free function